src/DotNetWorker.Core/DotNetWorker.Core.csproj

```diff
@@ -22,12 +22,27 @@
 
   <ItemGroup>
     <PackageReference Include="Azure.Core" Version="1.37.0" />
-    <PackageReference Include="Microsoft.Extensions.Hosting" Version="5.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="5.0.0" />
-    <PackageReference Include="System.Collections.Immutable" Version="5.0.0" />
-    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="7.0.0" />
   </ItemGroup>
 
+  <Choose>
+    <When Condition="'$(TargetFramework)' == 'net9.0'">
+      <ItemGroup>
+        <PackageReference Include="Microsoft.Extensions.Hosting" Version="9.0.0-preview.*" />
+        <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="9.0.0-preview.*" />
+        <PackageReference Include="System.Collections.Immutable" Version="9.0.0-preview.*" />
+        <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="9.0.0-preview.*" />
+      </ItemGroup>
+    </When>
+    <Otherwise>
```

Comment:
Should we make this explicit for each TFM? If doing so, I believe we can also just apply the condition to the item group directly. There's actually an example of that right below the current change.

Reply:
Similar feedback above. I'm assuming @kshyju was trying to avoid another evaluation for the fallback (block below), but I still think that will be simpler.

Reply:
Moved condition to ItemGroup element instead of Choose-When pattern